src/compiler/core.ts

```diff
@@ -365,10 +365,10 @@ namespace ts {
 
     export let localizedDiagnosticMessages: Map<string> = undefined;
 
-    export function getLocaleSpecificMessage(message: string) {
-        return localizedDiagnosticMessages && localizedDiagnosticMessages[message]
-            ? localizedDiagnosticMessages[message]
-            : message;
+    export function getLocaleSpecificMessage(message: DiagnosticMessage) {
+        return localizedDiagnosticMessages && localizedDiagnosticMessages[message.key]

```

Comment:
For a message that is an empty string, you'll always use the non-localized message unless you use `message.key in localizedDiagnosticMessages`. Though, that would be the most ridiculous edge case I may have ever brought up.